zebra-consensus/src/config.rs

```diff
@@ -30,9 +35,37 @@ pub struct Config {
     /// For security reasons, this option might be deprecated or ignored in a future Zebra
     /// release.
     pub checkpoint_sync: bool,
+}
+
+impl From<InnerConfig> for Config {
+    fn from(
+        InnerConfig {
+            checkpoint_sync, ..
+        }: InnerConfig,
+    ) -> Self {
+        Self { checkpoint_sync }
+    }
+}
+
+impl From<Config> for InnerConfig {
+    fn from(Config { checkpoint_sync }: Config) -> Self {
+        Self {
+            checkpoint_sync,
+            ..Default::default()
+        }
+    }
+}
+
+/// Inner consensus configuration for backwards compatibility with fields that have been removed.  
+#[derive(Clone, Debug, Eq, PartialEq, Default, Deserialize, Serialize)]
+#[serde(deny_unknown_fields, default)]
+pub struct InnerConfig {
```

Comment:
I'm not sure that an inner config is necessary, putting `#[serde(skip)]` on the field should do the same thing in a single line?
https://serde.rs/field-attrs.html#skip

(I would like to avoid reviewing this complex code change, because adding a single attribute is much easier to review.)

Reply:
Ah, I see what's happening here, we want Rust API users to not have the field, but toml configs to ignore it. That makes sense.